cpp-ch/local-engine/Rewriter/ExpressionRewriter.h

```diff
@@ -85,6 +85,24 @@ class GetJsonObjectFunctionWriter : public RelRewriter
             }
         }
     }
+    String getJsonPathOfGetJSONObject(const substrait::Expression_ScalarFunction & func)
+    {
+        String json_path = "";
+        for (size_t i = 1; i < func.arguments().size(); ++i)
+        {
+            auto json_path_pb = func.arguments(i).value();
+            if (!json_path_pb.has_literal() || !json_path_pb.literal().has_string())
+            {
+                break;
+            }
+            json_path += json_path_pb.literal().string();
+            if (i != func.arguments().size() - 1)
+            {
+                json_path += "#" ;
+            }
+        }
+        return json_path;
+    }
```

Comment:
When things become complex, I don't think this representation could be flexible enogh any more. Let's consider the case with three levels of nested calls. A tree structure should be OK. For example a json text as following
```json
[
  "path1",
  "path2",
  "path3":[
     "path3_1",
     "path3_2": [ ....]
  ],
  ...
]
```

Reply:
How about more levels of netsted calls here? The disccussion should be given

Reply:
If more levels of nested `get_json_object`call mixed with other `get_json_object` call, like `get_json_object(get_json_object(get_json_object(d, '$.a'), '$.b'), '$.c'), get_json_object(d, '$.e') ` and this will be optimized to `get_json_object(d, '$.a', '$.b', '$.c'), get_json_object(d, '$.e')` by the optimized rule `CollapsedNestedExpressions`, at next steps, the path of them will be converted to '$.a#$.b#$.c | $.e',  and it will be executed by `SparkFunctionGetJsonObject` to return a tuple eventually, So more levels case can be handled here.  @lgbo-ustc